app/furniture/marketplace/delivery_area.rb

```diff
@@ -4,8 +4,8 @@ class DeliveryArea < Record
     location(parent: :marketplace)
 
     belongs_to :marketplace, inverse_of: :delivery_areas
-    has_many :orders, inverse_of: :delivery_area
-    has_many :carts, inverse_of: :delivery_area
+    has_many :orders, -> { checked_out }, inverse_of: :delivery_area
+    has_many :carts, inverse_of: :delivery_area, dependent: :nullify
```

Comment:
Ah, and this works because `Marketplace::Carts` has a default scope of `:pre_checkout`, right? Nice

Reply:
Yes, which is part of why I kinda-like the default scopes; but also I now have a ticket for the refactor: https://github.com/zinc-collective/convene/issues/2020